docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,303 @@
+              model_engine.backward(loss)
+              model_engine.step()
+
+In this code snippet, ``core_context`` is the :class:`~determined.core.Context` instance which was
+initialized with :func:`determined.core.init`. The context manager requires access to both
+``core_context`` and the current :class:`~determined.core.SearcherOperation` instance (``op``) to
+appropriately report results.
+
+A full example which uses DeepSpeed Autotune with Core API can be found in the
+``examples/deepspeed_autotune/torchvision/core_api`` `subdirectory within the Determined GitHub Repo
+<https://github.com/determined-ai/determined/tree/master/examples/deepspeed_autotune/torchvision/core_api>`__
+.
+
+HuggingFace Trainer
+===================
+
+You can also use Determined's DeepSpeed Autotune with the HuggingFace (HF) Trainer and Determined's
+:class:`~determined.integrations.huggingface.DetCallback` callback object to optimize your DeepSpeed
+parameters.
+
+Similar to the previous case, you need to add a ``deepspeed_config`` field to the
+``hyperparameters`` section of your experiment configuration file, specifying the relative path to
+the DS ``json`` config file.
+
+Reporting results back to the Determined master requires both the ``dsat.dsat_reporting_context``
+context manager and ``DetCallback``.
+
+Furthermore, since ``dsat`` performs a search over different batch sizes and HuggingFace expects
+parameters to be specified as command-line arguments, an additional helper function,
+:func:`~determined.pytorch.dsat.get_hf_args_with_overwrites`, is needed to create consistent
+HuggingFace arguments.
+
+Here is an example code snippet from a HuggingFace Trainer script that contains key pieces of
+relevant code:
+
+.. code:: python
+
+   from determined.integrations.huggingface import DetCallback
+   from determined.pytorch import dsat
+   from transformers import HfArgumentParser,Trainer, TrainingArguments,
+
+   hparams = self.context.get_hparams()
+   parser = HfArgumentParser(TrainingArguments)
+   args = sys.argv[1:]
+   args = dsat.get_hf_args_with_overwrites(args, hparams)
+   training_args = parser.parse_args_into_dataclasses(args, look_for_args_file=False)
+
+   det_callback = DetCallback(core_context, ...)
+   trainer = Trainer(args=training_args, ...)
+   with dsat.dsat_reporting_context(core_context, op=det_callback.current_op):
+       train_result = trainer.train(resume_from_checkpoint=checkpoint)
+
+.. important::
+
+   -  The ``dsat_reporting_context`` context manager shares the same initial
+      :class:`~determined.core.SearcherOperation` as the ``DetCallback`` instance through its
+      ``op=det_callback.current_op`` argument.
+
+   -  The entire ``train`` method of the HuggingFace trainer is now wrapped in the
+      ``dsat_reporting_context`` context manager.
+
+A full example which uses DeepSpeed Autotune with HuggingFace Trainer can be found in the
+``examples/hf_trainer_api`` `subdirectory within the Determined GitHub Repo
```

Comment:
same as before, I would replace "subdirectory" with "examples"

Reply:
yeah, going to reorganize all three of these